### PR TITLE
ID:FPCO-15460;HOURS:5;DONE:100; Retry syncing when connection lost

### PR DESCRIPTION
**Pull request by ManishPrajapati-GoFynd**

### CODE REVIEW

File: src/lib/api/helper/interceptors.ts [hunk 1]
```diff
@@ -111,22 +118,39 @@ export function responseInterceptor() {
...
-            throw new Error(
-                'Not received response from the server, possibly some network issue, please retry!!'
-            );
+            
+            const originalRequest = error.config;
+            // Ask: Can I add custom header for this count?
+            // Check if it's network related error
+            // If yes, then add counter inside header
+            // If count is less than max retry limit, then resend request
+            if (["ECONNABORTED", "EPIPE", "ENOTFOUND", "ETIMEDOUT", "ECONNRESET"].includes(error.code)) {
```
Comment (vivek8690) [inline src/lib/api/helper/interceptors.ts:135]: Can't we use an inbuilt package like here  https://www.npmjs.com/package/axios-retry?
@manishprajapati52

**Review by vivek8690 — CHANGES_REQUESTED** [review]
Please look at the comments

**Review by vivek8690 — CHANGES_REQUESTED** [review]
Please look at the comments

File: src/lib/api/helper/interceptors.ts (hunk 1 above)
Comment (ManishPrajapati-GoFynd) [reply src/lib/api/helper/interceptors.ts:135]: As we discussed, please confirm, whether we want to use third party package or our own implementation?

Comment (vivek8690) [reply src/lib/api/helper/interceptors.ts:135]: Please go-ahead with library @manishprajapati52

**Pull request merged**